astropy/utils/data.py

```diff
@@ -1051,7 +1051,7 @@ def _download_file_from_source(source_url, show_progress=True, timeout=None,
                         except OSError:
                             pass
                     raise
-    return f.name, hasher.hexdigest()
+    return f.name
```

Comment:
I think given the problems people are having on clusters, getting it into 4.0.x should be a priority, even if it forces us to leave some fossil stuff in the API.

Reply:
Okay, let's see if we can reach a middle ground. I am open to discussions and ideas. Thanks again for your patience! 😄

Reply:
I'll put this back, but this *is* a hidden function, explicitly not included in `__all__` and flagged with a leading underscore, so I think it's unfair to consider this part of the API.

Reply:
In fact there's a real question here: for this internal function, do we go to the totally unnecessary extra effort of hashing the downloaded file and returning the hash just for backward compatibility with code that pokes around in the internals?

Reply:
@aarchiba -- Sorry for the confusion. For "hidden" functions, in the past, we have changed API without a deprecation warning, but I am not sure about policy of backporting to LTS. I was hoping to hear from @bsipocz or @eteq . But in case we don't and I have to make the call, I would say, it is okay to change API here for "hidden" function because there is a strong justification to do so. However, public API definitely needs to be preserved as before.